### PR TITLE
remove share on 404

### LINKED ISSUE #52: Remove mounted shares that are no longer accessible
(probably because they were unshared)

### PR DESCRIPTION
**Pull request by JoGir**
resolves #52

### CODE REVIEW

**Comment by Rehwanne** [comment]
retest this please

**Comment by Rehwanne** [comment]
retest this please

**Comment by Rehwanne** [comment]
retest this please

**Comment by Rehwanne** [comment]
retest this please

**Comment by audax** [comment]
https://github.com/Qabel/qabel-android/issues/52#issuecomment-185743168
 > BoxExternal has the property isAccessible and BoxNavigation allows to detach an external based on it's name. 

If isAccessible is not used and needed, you may just remove it.

Also, while you are at it: Please split up the tests to make the intention of the test clearer. Rule of thumb: If a test name has an "And" in it, split it up!
(I know you did not write the original test, but you can improve it)

Apart from that the PR is fin

**Comment by audax** [comment]
Can one of the admins verify this patch?

File: qabelbox/src/main/java/de/qabel/qabelbox/storage/AbstractNavigation.java [hunk 1]
```diff
@@ -108,13 +110,23 @@ public String getName() {
...
-    protected File blockingDownload(String prefix, String name, BoxTransferListener boxTransferListener) throws QblStorageNotFound {
+    protected File blockingDownload(String prefix, String name, BoxTransferListener boxTransferListener) throws QblStorageException {
         File file = transferManager.createTempFile();
         int id = transferManager.download(prefix, name, file, boxTransferListener);
         if (transferManager.waitFor(id)) {
             return file;
         } else {
-            throw new QblStorageNotFound("File not found. Prefix: " + prefix + " Name: " + name);
+            try {
+                Exception ex = transferManager.lookupError(id);
```
Comment (audax) [inline qabelbox/src/main/java/de/qabel/qabelbox/storage/AbstractNavigation.java]: No need for the variable ex, you either just inspect the exception directly or throw it directly.

File: qabelbox/src/main/java/de/qabel/qabelbox/storage/AbstractNavigation.java [hunk 2]
```diff
@@ -345,6 +359,12 @@ public BoxObject getExternal(String name) throws QblStorageException {
                 }
             }
         }
+        if (referencesNotFound.size() > 0) {
```
Comment (audax) [inline qabelbox/src/main/java/de/qabel/qabelbox/storage/AbstractNavigation.java]: extract to a private method with a descriptive name

**Pull request merged**